src/main/java/uk/gov/digital/ho/hocs/search/application/aws/SearchListener.java

```
@@ -0,0 +1,95 @@
+                    caseDataService.deleteCorrespondent(request.getCaseUUID(), correspondentDeleted);
+                    break;
+                case CASE_TOPIC_CREATED:
+                    CreateTopicRequest createTopicRequest = objectMapper.readValue(request.getData(), CreateTopicRequest.class);
+                    caseDataService.createTopic(request.getCaseUUID(), createTopicRequest);
+                    break;
+                case CASE_TOPIC_DELETED:
+                    DeleteTopicRequest deleteTopicRequest = objectMapper.readValue(request.getData(), DeleteTopicRequest.class);
+                    caseDataService.deleteTopic(request.getCaseUUID(), deleteTopicRequest);
+                    break;
+                case SOMU_ITEM_CREATED:
+                    SomuItemDto createSomuItem = objectMapper.readValue(request.getData(), SomuItemDto.class);
+                    caseDataService.createSomuItem(request.getCaseUUID(), createSomuItem);
+                    break;
+                case SOMU_ITEM_UPDATED:
+                    SomuItemDto updateSomuItem = objectMapper.readValue(request.getData(), SomuItemDto.class);
+                    caseDataService.updateSomuItem(request.getCaseUUID(), updateSomuItem);
+                    break;
+                case SOMU_ITEM_DELETED:
+                    SomuItemDto deleteSomuItem = objectMapper.readValue(request.getData(), SomuItemDto.class);
+                    caseDataService.deleteSomuItem(request.getCaseUUID(), deleteSomuItem);
+                    break;
+                default:
+                    throw new ApplicationExceptions.InvalidEventTypeException(String.format("Unexpected value: %s", request.getType()), LogEvent.UNKNOWN_SEARCH_MESSAGE_TYPE);
+            }
+        } else {
+            throw new ApplicationExceptions.InvalidEventTypeException("Unexpected value: Type was null", LogEvent.NULL_SEARCH_MESSAGE_TYPE);
```

Fail fast, move this outside of the block and above...